### PR TITLE
[Orc8r] Orcl Improvements and Test framework changes

### LINKED ISSUE #6634: [Orc8r] Add APIs to create and destroy cluster

### PR DESCRIPTION
**Pull request by karthiksubraveti**
Signed-off-by: Karthik Subraveti <ksubraveti@fb.com>

### gateway registration sanity test

`go test gateway_registration_test.go -ssh-key /root/project/keyMagmaHostDelta.pem  -certs-dir /root/project/secrets -cluster-config /root/project/cluster.json  -bastion-host 54.149.4.82
`
Was able to successfully register 50 gateways to orc8r 

### Test framework changes
Added AWS test cluster library which can
- bring up orc8r and gateways based on provided template. Added two templates in /root/data, default and scale template. Scale template brings up 50 gateways
- takes care of registering the orc8r domain in the parent domain's hosted zone
```
test-cluster create --template <template_file>(default template is chosen if template flag isn't provided)
test-cluster cleanup
```

### Cloudstrapper AGW testframework change
- Added changes to configure AGW instances through jumpnode (generate ssh configuration)
- Fixed ansible to run cloudformation in parallel to bring up and teardown all AGW instances
- Added changes to run node exporter within the individual gateway instances
- Added changes to run avalanche to simulate metrics on the gateway
- 
### Orcl Fixes
1. Added magma-uuid to all resources deployed by Orc8r. (ENIs and EKS created SG are the only exception at this point, we can reference everything else through magma-uuid)
2. Added artifactory repo as our default docker repo.
3. Minor fixes to orcl and associated playbooks 

### Terraform changes
1. Added support for cloudwatch exporter to export cloudwatch metrics to prometheus. It can be enabled by turning on `cloudwatch_exporter_enabled` terraform variable to true.

## Test Plan
Existing tests ran fine
```
ksubraveti@ksubraveti-mbp docker % ./run_deployer.bash --deploy-dir ~/workspace/orc8r_15_deployment_7 --test check_all
Build false
Deploy workddir /Users/ksubraveti/workspace/orc8r_15_deployment_7
Warning: inferring magma root(/Users/ksubraveti/workspace/magma) from run_deployer script directory
check_all
Running orcl container unit tests

Configuring infra deployment variables

Configuring platform deployment variables

Configuring service deployment variables
.
Configuring platform deployment variables
Setting key deploy_elastic value true for component platform
.
----------------------------------------------------------------------
Ran 2 tests in 0.028s

OK
Checking if helm charts are in sync

helm charts are in sync
Checking tf vars are in sync
vars.yml found to be in sync with terraform modules
```

Most resources are tagged magma-uuid
```
root@232372b27609:~/project# aws resourcegroupstaggingapi get-resources  --tag-filters "Key=magma-uuid,Values=default" | grep ResourceARN
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:elastic-ip/eipalloc-0387832726094186c",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:instance/i-0699e35ac6753a733",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:instance/i-05cf71fbcb3f3d1ed",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:instance/i-00c0790108a0f2df0",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:instance/i-09783419916ed352f",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:internet-gateway/igw-06072f9ad6f8e5114",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:natgateway/nat-0edf57b8b8c6b630c",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:natgateway/nat-048f1aad888ad5684",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:route-table/rtb-0d70a79c2a9770dd8",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:route-table/rtb-00ebb65e2501795c2",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-0de0b6955c03fcef7",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-01866f9dd42945c06",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-0bf2f6852b3dfefc7",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-00ce42ca7d934f19e",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-0cc5410ea84d8c560",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-0635fc73ae21794f9",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-008b8914e9898cb57",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-0b2681f52966555c8",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-013aed8e3f29e2adb",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-0739901efffb9e5ad",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-00d88cc19d78f946e",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-01e16c052a1460bf6",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-08b453eea0dd23f91",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-0ee8ef6d320021d8c",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-062f9a6eae7bade11",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-014e732bba92eb7ea",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:subnet/subnet-07c92461e8ecbaec8",
            "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:vpc/vpc-0bf4edaa088366482",
            "ResourceARN": "arn:aws:eks:us-west-2:063307678321:cluster/orc8r",
            "ResourceARN": "arn:aws:elasticfilesystem:us-west-2:063307678321:file-system/fs-beccc8b9",
            "ResourceARN": "arn:aws:elasticloadbalancing:us-west-2:063307678321:loadbalancer/aa38cb258e4254f93884270eaa0c4f60",
            "ResourceARN": "arn:aws:elasticloadbalancing:us-west-2:063307678321:loadbalancer/a1181aead1db14d17ad29bd7c13014fb",
            "ResourceARN": "arn:aws:elasticloadbalancing:us-west-2:063307678321:loadbalancer/a12f348107151481d8d35a43a574e8fd",
            "ResourceARN": "arn:aws:elasticloadbalancing:us-west-2:063307678321:loadbalancer/a22c847e3a318438d935dc2bed6b08f7",
            "ResourceARN": "arn:aws:elasticloadbalancing:us-west-2:063307678321:loadbalancer/af51bb0cbd5b14f2c988e431cc87239b",
            "ResourceARN": "arn:aws:es:us-west-2:063307678321:domain/orc8r-es",
            "ResourceARN": "arn:aws:logs:us-west-2:063307678321:log-group:/aws/eks/orc8r/cluster",
            "ResourceARN": "arn:aws:rds:us-west-2:063307678321:subgrp:orc8r_vpc",

The ones missing are ENIs and default security group
root@232372b27609:~/project# aws resourcegroupstaggingapi get-resources  | grep ResourceARN | sort > /tmp/1
root@232372b27609:~/project# aws resourcegroupstaggingapi get-resources  --tag-filters "Key=magma-uuid,Values=default" | grep ResourceARN | sort > /tmp/2
root@232372b27609:~/project# diff /tmp/1 /tmp/2
9,15d8
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:network-interface/eni-033b0268d6b8b8df3",
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:network-interface/eni-0404e4e4961fce969",
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:network-interface/eni-06bd5c8e433da9e1f",
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:network-interface/eni-0918de58abaf3fbab",
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:network-interface/eni-0c3c3778fbac8ece5",
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:network-interface/eni-0da8633c4a28d1bbe",
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:network-interface/eni-0e825b0659ac6f57d",
21d13
<             "ResourceARN": "arn:aws:ec2:us-west-2:063307678321:security-group/sg-0428a93cec8e5a199",

For EKS created SG - https://github.com/aws/containers-roadmap/issues/374#issuecomment-775251852, still waiting on tag propagation

```
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by themarwhal** [comment]
Hey @karthiksubraveti there have been some recent CODEOWNERS file changes, so please rebase onto master when putting this out for review. Thanks! :)

**Review by hcgatewood — APPROVED** [review]
Stamping to unblock sharing, will further review once we stabilize

**Pull request merged**
